core/app/[locale]/(default)/account/(tabs)/orders/page-data.ts

```diff
@@ -0,0 +1,369 @@
+          }
+          billingAddress {
+            firstName
+            lastName
+            address1
+            city
+            stateOrProvince
+            postalCode
+            country
+          }
+          consignments {
+            shipping(before: $before, after: $after, first: $first, last: $last) {
+              edges {
+                node {
+                  entityId
+                  shippingAddress {
+                    firstName
+                    lastName
+                    address1
+                    city
+                    stateOrProvince
+                    postalCode
+                    country
+                  }
+                  shipments {
+                    edges {
+                      node {
+                        entityId
+                        shippedAt {
+                          utc
+                        }
+                        ...OrderShipmentFragment
+                      }
+                    }
+                  }
+                  lineItems {
+                    edges {
+                      node {
+                        ...OrderItemFragment
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `,
+  [OrderItemFragment, OrderShipmentFragment],
+);
+
+const ProductAttributes = graphql(`
+  query ProductAttributes($entityId: Int) {
+    site {
+      product(entityId: $entityId) {
+        path
+      }
+    }
+  }
+`);
+
+type Variables = VariablesOf<typeof ProductAttributes>;
+
+export const getProductAttributes = cache(async (variables: Variables) => {
+  const response = await client.fetch({
+    document: ProductAttributes,
+    variables,
+    fetchOptions: { next: { revalidate } },
+  });
+
+  return response.data.site.product;
+});
+
+type ShippingConsignments = NonNullable<
+  NonNullable<ResultOf<typeof CustomerOrderDetails>['site']['order']>['consignments']
+>['shipping'];
+
+const addProductAttributesToShippingConsignments = async (consignments: ShippingConsignments) => {
+  const shipping = removeEdgesAndNodes(consignments);
+
+  const shippingConsignments = await Promise.all(
+    shipping.map(async (consignment) => {
+      const { lineItems, shipments, ...otherItems } = consignment;
+      const extendedLineItems = await Promise.all(
+        removeEdgesAndNodes(lineItems).map(async ({ productEntityId, ...otherAttributes }) => {
+          const productAtrributes = await getProductAttributes({
+            entityId: productEntityId,
+          });
+
+          const { path = '' } = productAtrributes ?? {};
+
+          return {
+            productEntityId,
+            path,
+            ...otherAttributes,
+          };
+        }),
+      );
+
+      return {
+        lineItems: extendedLineItems,
+        shipments: removeEdgesAndNodes(shipments),
+        ...otherItems,
+      };
+    }),
+  );
+
+  return shippingConsignments;
+};
```

Comment:
🍹 This page is going to be really slow with a large amount of `consignments` & `lineItems` due to the `O(n^2)` amount of promises happening here. We probably need to refactor this to be more efficient. Maybe we can use some suspense boundaries instead to fetch while the page is rendered.

Reply:
already fixed